server/server/src/server.ts

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright © 2021 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { createServer, IncomingMessage } from 'http'
+import WebSocket, { AddressInfo, Server } from 'ws'
+import { Storage, Tx } from '@anticrm/core'
+
+export interface ServerProtocol {
+  shutdown: () => void
+  address: () => { host: string; port: number }
+}
+export interface StorageProvider {
+  connect: (clientId: string, tx: (tx: Tx) => void) => Storage
+  close: (clientId: string) => void
+}
+
+export async function start (
+  host: string | undefined,
+  port: number,
+  provider: StorageProvider
+): Promise<ServerProtocol> {
+  console.log(`starting server on port ${port}...`)
+  console.log(`host: ${host ?? 'localhost'}`)
+
+  const server = createServer()
+  const wss = new Server({ noServer: true })
+
+  const connections = new Map<string /* clientId */, WebSocket>()
+
+  wss.on('connection', (ws: WebSocket, request: any, clientId: string) => {
+    console.log('connect:', clientId)
+
+    const storage = provider.connect(clientId, tx => {
+      // Send transaction to client.
+      ws.send(
+        JSON.stringify({
+          _id: tx._id,
+          tx: tx
+        })
+      )
```

Comment:
Main question for all this initial server implementation: _why not JsonRPC?_ (`packages/rpc`)

Reply:
We could do so, not sure we need to put it inside packages/rpc as it was before.

Found following specs: https://www.jsonrpc.org/specification

Reply:
I know the spec, let's get `packages/rpc` back from old repo -- it was used everywhere :)

Reply:
I think the one from `v0.4` is fine